### ISSUE TITLE
installhtml installs private pod

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11859]
Migrated from [rt.perl.org#107880](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=107880) (status was 'open')

Searchable as RT107880$

**Comment by p5pRT** [comment]
### From @nwc10

\./installhtml is installing pod that it isn't meant to\.

$ cd /home/nick/Sandpit/snap\-v5\.15\.6\-450\-g6ca15f9/lib/perl5/5\.15\.6/html
$ ls \-l lib/XS/\*\.html Porting/\*
\-rw\-r\-\-r\-\- 1 nick nick    847 Jan 10 18&#8203;:21 Porting/GitUtils\.html
\-rw\-r\-\-r\-\- 1 nick nick 105825 Jan 10 18&#8203;:21 Porting/epigraphs\.html
\-rw\-r\-\-r\-\- 1 nick nick  12274 Jan 10 18&#8203;:21 Porting/how\_to\_write\_a\_perldelta\.html
\-rw\-r\-\-r\-\- 1 nick nick  15160 Jan 10 18&#8203;:21 Porting/perldelta\_template\.html
\-rw\-r\-\-r\-\- 1 nick nick  49944 Jan 10 18&#8203;:21 Porting/release\_managers\_guide\.html
\-rw\-r\-\-r\-\- 1 nick nick   5635 Jan 10 18&#8203;:21 Porting/release\_schedule\.html
\-rw\-r\-\-r\-\- 1 nick nick   9614 Jan 10 18&#8203;:30 lib/XS/APItest\.html
\-rw\-r\-\-r\-\- 1 nick nick   2477 Jan 10 18&#8203;:30 lib/XS/Typemap\.html

None of these should be installed\, given that installperl installs none of
their raw Pods\, and installman installs none of them as man pages\.

Probably installhtml should be refactored to use the same exclusion lists
as installman\, by calling pods\_to\_install\(\) in Porting/pod\_lib\.pl

I don't know if this is a regression from perl 5\.14\.0

Nicholas Clark

**Comment by p5pRT** [comment]
### From @nwc10

On Tue Jan 10 11&#8203;:28&#8203;:11 2012\, nicholas wrote&#8203;:
> \./installhtml is installing pod that it isn't meant to\.
> 
> $ cd /home/nick/Sandpit/snap\-v5\.15\.6\-450\-
> g6ca15f9/lib/perl5/5\.15\.6/html
> $ ls \-l lib/XS/\*\.html Porting/\*

and TestInit\.html\, from TestInit\.pm

Likely this is all a side effect of it scanning from the top of the
distribution tree\, instead of from lib/

Nicholas Clark

**Comment by p5pRT** [comment]
@nwc10 - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

Another potential problem with 'installhtml' has been described in #12257

**Comment by jkeenan** [comment]
Back in 2012, @nwc10 was reporting that the following files should not have been installed:
```
$ ls -l lib/XS/*.html Porting/*
-rw-r--r-- 1 nick nick 847 Jan 10 18​:21 Porting/GitUtils.html
-rw-r--r-- 1 nick nick 105825 Jan 10 18​:21 Porting/epigraphs.html
-rw-r--r-- 1 nick nick 12274 Jan 10 18​:21 Porting/how_to_write_a_perldelta.html
-rw-r--r-- 1 nick nick 15160 Jan 10 18​:21 Porting/perldelta_template.html
-rw-r--r-- 1 nick nick 49944 Jan 10 18​:21 Porting/release_managers_guide.html
-rw-r--r-- 1 nick nick 5635 Jan 10 18​:21 Porting/release_schedule.html
-rw-r--r-- 1 nick nick 9614 Jan 10 18​:30 lib/XS/APItest.html
-rw-r--r-- 1 nick nick 2477 Jan 10 18​:30 lib/XS/Typemap.html
```
and
```
and TestInit.html, from ./TestInit.pm
```
Attempting to understand this problem (I don't customarily use `installhtml` for anything), this evening I installed blead into $TESTINGDIR, then used this command:
```
perl $SECONDARY_CHECKOUT_DIR/installhtml \
  --podroot=$TESTINGDIR/blead     \
  --htmldir=/tmp/gh-11859-html     \
  --htmlroot=/tmp/gh-11859-html    \
  --recurse \
  --verbose
```
I got 858 .html files installed under /tmp/gh-11859-html/lib/perl5/5.33.7.  But I didn't get any among those Nick listed.

Have I invoked `installhtml` plausibly?  If so, can we close this ticket on the assumption that the problem was fixed somewhere along the line?

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
Should be addressed by https://github.com/Perl/perl5/pull/19075.

**Comment by jkeenan** [comment]
> Should be addressed by #19075.

Am taking this ticket for the purpose of closing it within 7 days unless we hear otherwise from @nwc10 or others.

**Comment by jkeenan** [comment]
> > Should be addressed by #19075.
> 
> Am taking this ticket for the purpose of closing it within 7 days unless we hear otherwise from @nwc10 or others.

**CORRECTION:** There's a [pull request](https://github.com/Perl/perl5/pull/19075) needing review for this issue.

**Comment by jkeenan** [comment]
> > > Should be addressed by #19075.
> > 
> > 
> > Am taking this ticket for the purpose of closing it within 7 days unless we hear otherwise from @nwc10 or others.
> 
> **CORRECTION:** There's a [pull request](https://github.com/Perl/perl5/pull/19075) needing review for this issue.

The code for https://github.com/Perl/perl5/pull/19075 has been merged into blead.  That should resolve this ticket.

Thank you very much.
Jim Keenan
